matchzoo/engine/base_generator.py

```diff
@@ -20,26 +20,20 @@ def __init__(
         self,
         batch_size: int = 32,
         num_instances: int = 0,
-        stage: str = 'train',
         shuffle: bool = True
     ):
         """
         :class:`BaseGenerator` constructor.
 
         :param batch_size: number of instances for each batch
         :param num_instances: total number of instances
-        :param stage: String indicate the pre-processing stage, `train`,
-            `evaluate`, or `predict` expected.
         :param shuffle: a bool variable to determine whether choose samples
         randomly
         """
-        self.batch_size = batch_size
-        self.shuffle = shuffle
-        self.num_instances = num_instances
-        if stage not in ['train', 'evaluate', 'predict']:
-            raise ValueError(f'{stage} is not a valid stage name.')
-        self.stage = stage
-        self.index_array = None
+        self._batch_size = batch_size
+        self._shuffle = shuffle
+        self._num_instances = num_instances
+        self._index_array = None
```

Comment:
can we use `indices` instead of `index_array`?

Reply:
I like `indices` too, but it was `index_array` and I thought it's not too bad.